app/src/controllers/userController.ts

```diff
@@ -0,0 +1,127 @@
+        try {
+            const users = await User.find();
+            res.json({
+                statusCode: 200,
+                data: users
+            });
+        } catch(err) {
+            res.json({
+                statusCode: 400,
+                errorMessage: err.message || err.toString()
+            });
+        }
+    }
+
+    /**
+     * creates a new user instance
+     * @param  {Request} req - express request object with updated user details in body
+     * @param  {Response} res — express response object
+     * @return json object with statusCode and and created user
+     */
+    public async create(req: Request, res: Response) {
+        try {
+            let newUser = new User({
+                name: req.body.name,
+                email: req.body.email,
+            });
+            const user = await newUser.save();
+            res.json({
+                statusCode: 200,
+                data: user
+            });
+        } catch(err) {
+            res.json({
+                statusCode: 400,
+                errorMessage: err.message || err.toString()
+            });
+        }
+    }
+
+    /**
+     * displays details of specific user instance
+     * @param  {Request} req - express request object with userId parameter
+     * @param  {Response} res — express response object
+     * @return json object with statusCode and user details
+     */
+    public async find(req: Request, res: Response) {
+        try {
+            const user = await User.findById(req.params.userId);
+            res.json({
+                statusCode: 200,
+                data: user
+            });
+        } catch(err) {
+            res.json({
+                statusCode: 400,
+                errorMessage: err.message || err.toString()
+            });
+        }
+    }
+
+    /**
+     * updates a specific user instance
+     * @param  {Request} req - express request object with userId parameter and new user attributes
+     * @param  {Response} res — express response object
+     * @return json object with statusCode and updated user details
+     */
+    public async update(req: Request, res: Response) {
+        try {
+            const user = await User.findOneAndUpdate({
+                _id: req.params.userId
+            }, {
+                name: req.body.name,
+                email: req.body.email,
+            }, { new: true });
+            res.json({
+                statusCode: 200,
+                data: user
+            });
+        } catch(err) {
+            res.json({
+                statusCode: 400,
+                errorMessage: err.message || err.toString()
+            });
+        }
+    }
+
+    /**
+     * deletes a specific user instance
+     * @param  {Request} req - express request object with userId parameter
+     * @param  {Response} res — express response object
+     * @return json object with statusCode
+     */
+    public async delete(req: Request, res: Response) {
```

Comment:
Unwanted method

Reply:
If it's ok I would leave it as an example for other models we will need a delete method on and later delete or disable this method?
I also think that it would be useful in the long run when we will have an administration interface for the service.